runtime/compiler/aarch64/codegen/CallSnippet.cpp

```diff
@@ -410,11 +410,18 @@ uint8_t *TR::ARM64VirtualUnresolvedSnippet::emitSnippetBody()
    TR_J9VMBase *fej9 = (TR_J9VMBase *)(comp->fe());
    void *thunk = fej9->getJ2IThunk(callNode->getSymbolReference()->getSymbol()->castToMethodSymbol()->getMethod(), comp);
 
+   // for alignment of intptr_t data
+   if (((uint64_t)cursor % TR::Compiler->om.sizeofReferenceAddress()) == 0)
+      {
```

Comment:
Since this code is here to align the data _after_ the call instruction, perhaps adding an assert that `sizeof(int32_t) == ARM64_INSTRUCTION_LENGTH` is warranted just to make it clear of the assumption you're making.

Reply:
I added `TR_ASSERT()` after `cursor += ARM64_INSTRUCTION_LENGTH;` in both ARM64VirtualUnresolvedSnippet and ARM64InterfaceCallSnippet.